pypsa/linopt.py

```diff
@@ -685,12 +691,22 @@ def run_and_read_gurobi(n, problem_fn, solution_fn, solver_logfile,
     Status = gurobipy.GRB.Status
     statusmap = {getattr(Status, s) : s.lower() for s in Status.__dir__()
                                                 if not s.startswith('_')}
-    status = statusmap[m.status]
-    termination_condition = status
-    if termination_condition != "optimal":
-        return status, termination_condition, None, None, None
-    else:
+    termination_condition = statusmap[m.status]
+
+    if termination_condition == "optimal":
         status = 'ok'
+    elif termination_condition == 'suboptimal':
+        status = 'warning'
+    elif termination_condition == "infeasible":
+        status = 'warning'
```

Comment:
I think these lines can be deleted or?

Reply:
Yes, but I'd rather keep them to show what possible termination conditions are possible, especially if we want to change the behaviour later.